crates/ruff_python_parser/src/lib.rs

```diff
@@ -355,44 +354,6 @@ impl Tokens {
         TokenKindIter::new(&self.0)
     }
 
-    /// Returns an iterator over the [`TokenKind`] and its range for all the tokens that are
-    /// within the given `range`.
-    ///
-    /// The start and end position of the given range should correspond to the start position of
-    /// the first token and the end position of the last token in the returned iterator.
-    ///
-    /// For example, if the struct contains the following tokens:
-    /// ```txt
-    /// (Def, 0..3)
-    /// (Name, 4..7)
-    /// (Lpar, 7..8)
-    /// (Rpar, 8..9)
-    /// (Colon, 9..10)
-    /// (Ellipsis, 11..14)
-    /// (Newline, 14..14)
-    /// ```
-    ///
-    /// Then, the range `4..10` returns an iterator which yields `Name`, `Lpar`, `Rpar`, and
-    /// `Colon` token. But, if the given position doesn't match any of the tokens, an empty
-    /// iterator is returned.
-    pub fn kinds_within_range<T: Ranged>(&self, ranged: T) -> TokenKindIter {
```

Comment:
It's possible @dhruvmanila has plans for this one — he added it two weeks ago in https://github.com/astral-sh/ruff/commit/025768d30344b275e9caa2f72a17db8fa4a63ccd

Reply:
Yeah, I'm using this in the new parser PRs. It's fine to remove it here as it has changed a bit, so I'll resolve it during the rebase.